### PR TITLE
Enhance JBoss EAP port detection

### LINKED ISSUE #195: Enhance Wildfly & EAP port detection
## Which area this feature is related to?

/area cli

## Issue details

After the implementation of [Wildfly and JBoss EAP detectors](https://github.com/redhat-developer/alizer/pull/161) we need to improve the port detection for these cases.

The first attempt to do that, was to detect a `standalone.xml` file and check its contents (see more info [here](https://github.com/redhat-developer/alizer/pull/161#discussion_r1158210346)). But as this file is produced after a maven build we are not able to detect ports from it.

Another workaround could be to check the `pom.xml` file for any scripts that are might configuring the server. An example inside pom.xml could be:
```xml
<packaging-scripts>
    <packaging-script>
      <scripts>
        <script>../scripts/script1.cli</script>
      </scripts>
    </packaging-script>
<packaging-scripts>
```
An example line that could configure a port could be (Check more information [here](https://access.redhat.com/documentation/en-us/red_hat_jboss_enterprise_application_platform/7.0/html/configuration_guide/network_and_port_configuration#socket_bindings)):
```
/socket-binding-group=new-sockets/socket-binding=new-socket-binding:add(port=1234)
```

Example project using cli file: https://github.com/wildfly/quickstart/tree/main/servlet-security

**Target Date**: 26 June 2023

### PR DESCRIPTION
**Pull request by thepetk**
## What does this PR do?

This PR follows the instructions given in [alizer#195-comment](https://github.com/redhat-developer/alizer/issues/195#issuecomment-1599042457) in order to detect any configured ports for `JBoss EAP` detected components.

As a result, when we identify a `configuration > javaOpts` field inside `pom.xml`, we are trying to match the `json.http.port` for and get the port value configured in the plugin.

### Which issue(s) does this PR fix

Partially fixes #195 

### PR acceptance criteria

Testing and documentation do not need to be complete in order for this PR to be approved. We just need to ensure tracking issues are opened.

- [x] Unit/Functional tests

  

- [x] Documentation

### CODE REVIEW

File: go/pkg/utils/detector.go [hunk 1]
```diff
@@ -149,6 +150,21 @@ func GetPomFileContent(pomFilePath string) (schema.Pom, error) {
 	return pom, nil
 }
 
+// GetJbossCLIFileContent returns the pom found in the path.
+func GetJbossCLIFileContent(jbossCliFilePath string) (schema.JBossCLIXML, error) {
```
[the remark this answers is not in the document]
Comment (thepetk) [reply go/pkg/utils/detector.go:154]: I've added a separate issue for gosec (#222)

**Comment by thepetk** [comment]
@mike-hoang all gosec warnings have been fixed

File: go/pkg/apis/enricher/framework/java/jboss_eap_detector.go [hunk 2]
```diff
@@ -31,5 +33,23 @@ func (o JBossEAPDetector) DoFrameworkDetection(language *model.Language, config
 }
 
 func (o JBossEAPDetector) DoPortsDetection(component *model.Component, ctx *context.Context) {
-	// Not implemented
+	var ports []int
+	jbossCli := filepath.Join(component.Path, "src/main/resources/bin/jboss-cli.xml")
```
Comment (mike-hoang) [inline go/pkg/apis/enricher/framework/java/jboss_eap_detector.go:37]: I'm not familiar with JBoss at all so just curious if this is where the `jboss-cli.xml` will always be stored?

Comment (thepetk) [reply go/pkg/apis/enricher/framework/java/jboss_eap_detector.go:37]: >Certain aspects of the management CLI can be customized in its configuration file, jboss-cli.xml. This file must be located either in the EAP_HOME/bin directory or in a custom directory specified with the jboss.cli.config system property.

I've found this in the [documentation for Jboss configuration](https://access.redhat.com/documentation/en-us/red_hat_jboss_enterprise_application_platform/7.1/html/management_cli_guide/config_management_cli)

**Review by mike-hoang — APPROVED** [review]

**Pull request merged**